application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/DiagramContentHandler.java

```diff
@@ -0,0 +1,209 @@
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiAttachment;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Handler for diagram content operations.
+ * 
+ * @version $Id$
+ * @since 1.13
+ */
+public class DiagramContentHandler
+{
+    /**
+     * View action.
+     */
+    public static final String VIEW_ACTION = "view";
+
+    /**
+     * Pages referenced inside diagram content.
+     */
+    private List<String> includedPages = new ArrayList<String>();
+
+    /**
+     * Helper methods for custom links contained by a diagram.
+     */
+    private DiagramLinkHandler linkHandler;
+
+    /**
+     * Constructor.
+     */
+    public DiagramContentHandler()
+    {
+        this.linkHandler = new DiagramLinkHandler();
+    }
+
+    /**
+     * Update the attachments content and save the new version.
+     * 
+     * @param backlinkDoc document that has a backlink to diagram
+     * @param context the execution context
+     * @param oldDocRef reference of the page before rename
+     * @param newDocRef reference of the page after rename
+     * @throws XWikiException when an error occurs while accessing documents.
+     * @throws IOException if something goes wrong while setting attachment's content.
+     */
+    public void updateAttachment(XWikiDocument backlinkDoc, XWikiContext context, DocumentReference oldDocRef,
+        DocumentReference newDocRef) throws XWikiException, IOException
+    {
+        XWikiAttachment attachment = backlinkDoc.getAttachment("diagram.svg");
+        if (attachment != null) {
+            backlinkDoc.addAttachment(getUpdatedAttachment(attachment, context, oldDocRef, newDocRef));
+            context.getWiki().saveDocument(backlinkDoc, "Updated attachment after page rename", context);
+        }
+    }
+
+    /**
+     * Migrate links inside diagram's attachment to the new name.
+     * 
+     * @param attachment attachment to be processed
+     * @param context the execution context
+     * @param oldDocRef reference of the page before rename
+     * @param newDocRef reference of the page after rename
+     * @return the modified attachment
+     * @throws XWikiException when an error occurs while accessing documents.
+     * @throws IOException if something goes wrong while setting attachment's content.
+     */
+    public XWikiAttachment getUpdatedAttachment(XWikiAttachment attachment, XWikiContext context,
+        DocumentReference oldDocRef, DocumentReference newDocRef) throws XWikiException, IOException
+    {
+        XWikiDocument oldDoc = context.getWiki().getDocument(oldDocRef, context);
+        XWikiDocument newDoc = context.getWiki().getDocument(newDocRef, context);
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(attachment.getContentInputStream(context)));
+        StringBuffer sb = new StringBuffer();
+
+        String str;
+        while ((str = reader.readLine()) != null) {
+            sb.append(str);
+        }
```

Comment:
You should use ``IOUtils`` to read the attachment content into a string, in a single line of code.